packages/ui/src/models/visualization/flows/support/camel-component-schema.service.ts

```diff
@@ -42,71 +72,106 @@ export class CamelComponentSchemaService {
     }
   }
 
-  static getProcessorStepsProperties(
-    processorName: string,
-  ): { name: string; type: 'single-processor' | 'steps-list' | 'expression-list' }[] {
+  static canHavePreviousStep(processorName: keyof ProcessorDefinition): boolean {
+    return !this.DISABLED_SIBLING_STEPS.includes(processorName);
+  }
+
+  static getCompatibleComponents(processorName: keyof ProcessorDefinition): CatalogFilter {
     switch (processorName) {
+      case 'choice':
+        return {
+          kinds: [CatalogKind.Processor],
+          names: ['when', 'otherwise'],
+        };
+
+      case 'doTry':
+        return {
+          kinds: [CatalogKind.Processor],
+          names: ['doCatch', 'doFinally'],
+        };
+
+      default:
+        return {};
+    }
+  }
+
+  static getProcessorStepsProperties(processorName: keyof ProcessorDefinition): CamelProcessorStepsProperties[] {
+    switch (processorName) {
+      /** choice */ case 'when':
+      /** choice */ case 'otherwise':
+      /** doTry */ case 'doCatch':
+      /** doTry */ case 'doFinally':
+      case 'aggregate':
+      case 'onFallback':
+      case 'saga':
+      case 'from' as keyof ProcessorDefinition:
+        return [{ name: 'steps', type: 'steps-list' }];
+
       case 'choice':
         return [
           { name: 'when', type: 'expression-list' },
           { name: 'otherwise', type: 'single-processor' },
         ];
 
-      case 'aggregate':
       case 'doTry':
-      case 'doCath':
-      case 'doFinally':
-      case 'onFallback':
-      case 'saga':
-      case 'from':
-      case 'when':
-      case 'otherwise':
-        return [{ name: 'steps', type: 'steps-list' }];
+        return [
+          { name: 'steps', type: 'steps-list' },
+          { name: 'doCatch', type: 'expression-list' },
```

Comment:
Could this `expression-list` be for example 'clause-list` instead? then `otherwise` could be `single-clause` maybe?

Reply:
Sure, I'll rename `expression-list` to `clause-list`.

For `otherwise`, the intent is not to signal that it's a `single-clause` but rather that this step can contain an array of steps, like `from` or `aggregate`, is there another name that could serve this purpose?

Reply:
Hmm then, how about `branch-list` vs. `branch`, where `when` under `choice` is a `branch-list`? It implies that the `branch` has `steps` as a child, including `aggregate`, `filter`, etc etc.
One thing to note here is that, `when` itself is also a `branch` with the terms above, where you can use `when` as a step alone without `choice`, not as an array of `WhenDefinition` but a single one,  just like `filter`. It's just `choice` who defines `when` as an array of `WhenDefinition`.

BTW, down the road we will need a different axis to group `expression-aware` steps, e.g. `when`, `filter`, `setBody`, `setHeader`, etc.

Reply:
Also note that we can dynamically/programatically create a list of `branch` steps (has `steps/ProcessorDefinition[]` as a child) from YAML DSL schema as well as a list of `expression-aware` steps (has `expression/ExpressionDefinition` as a child) in camel-catalog pre-processing. Since Camel DSL is matured, it's much less frequent to change, but it's still a peace of mind to reduce the source of the fact.

Reply:
Alrighty, after the discussion, I updated the names to better reflect the type of content that each property can hold.